### ISSUE TITLE
bundletool version 1.8.2 & 1.9.0 generated universal apk is 3 times the size of aab

### ISSUE THREAD

**Issue by apar945** [google/bundletool#272]
Hello 

I am unity the google play services plugin in unity to built an asset bundle that contains the main scene in the base asset and rest of the scenes and textures separate bundles. When I build the android app bundle, supporting API levels 26 and above, the size of the .aab file is 230 MB. 

I then used the most recent versions of the bundletool to generate the universal apk and found out the size of the apk is 660 MB. I notice the same behaviour in play console as well where the universal apk is of similar size. 

So my question is why is the apk uncompressed and 3 times the size of the aab? Why can't the bundletool generate at least the same size apk? This is contrary to the purpose of asset bundles to reduce the delivery size to the devices.

When generating build-apks argument, the apks files is 4 times the size and the base apk is 70% of that archive.  

Here is a snapshot of the apks archive

![Apks1](https://user-images.githubusercontent.com/48810983/158099187-b24e1784-caa2-4ede-8bb6-d30f6535c5e6.PNG)

![Apks2](https://user-images.githubusercontent.com/48810983/158099197-8f646adc-3939-4ce8-801a-95211935b566.PNG)

Here is a snapshot of the universal apk archive
![Apk](https://user-images.githubusercontent.com/48810983/158099308-15db695b-30e3-466f-b054-66c185e3d049.PNG)

Finally the .aab archive which does cmpress the base assets
![Aab](https://user-images.githubusercontent.com/48810983/158099502-41e09801-baae-4580-8f88-b9230f572dd8.PNG)

**Comment by ymakhno** [comment]
All entries inside AAB file are always compressed but inside APK there might be uncompressed files. Usually in games asset files are stored in uncompressed form to allow its raw access from the APK without uncompressing on the device.

For `install-time` asset packs there is an option to compress its content but I do not know exact place in Unity where you can request it. `fast-follow` and `on-demand` assets packs should always be uncompressed.

**Comment by apar945** [comment]
@ymakhno,
I have tried setting all assets to be install time and they are still not compressed. I would like to know how the bundletool can compress these.

**Comment by apar945** [comment]
Hello @plecesne 

The size on the device is not an issue. I agree that it should be uncompressed for faster access.
However, for a user to see that they are downloading 700 MB installer rather than the compressed 160 - 180 MB is not great. 
We have been using a mobile device management platform where the apk download size is 160 MB and now that I am testing it on the play store, through internal testing, I see the download size as 700  MB and it takes a lot longer.

Is this the reason why android app bundles replaced apk and obb files? ie to reduce the app download size.

**Comment by apar945** [comment]
Hello All,

Here is an update from google play support who had this issue open from March 2022 to Aug 2022 (as far as I know).
There has been an update in the Play Console and now the same file that I had uploaded in March 2022 is downloaded in its compressed form of 160 - 180 MB. Back in March 2022, this same download was 700+ MB.

I don't have the details of what was fixed but it seems like it was something at the Play Console end that resolved this.

**Comment by a-l-e-x-d-s-9** [comment]
@ymakhno @plecesne
I'm experiencing this exact problem right now, without making any major changes in my app, besides a minor Unity version upgrade, the bundle downloaded from Google used to be around 100MB, but now it suddenly jumped to 350MB+. The AAB that I uploaded is just 130MB, and I haven't added more assets.
I'm using Google play-unity-plugins v1.7.0 for months, I think it comes with bundletool 1.7. Recently I started using Target API 32, with Unity 2020.3.40f1/41f1, then the problem appeared a couple of weeks ago, but I have noticed this problem only now.
@apar945 Can you please point me to Google play support case that you mentioned?

**Comment by apar945** [comment]
@a-l-e-x-d-s-9 
This is a later reply but hopefully it helps. The case with google play is not an open case. I logged the issue by emailing
googleplay-developer-support@google.com
The issue was resolved in November 2022.
Since then I have upgraded to Unity 2021 and the bundle built from the latest version also experienced the same issue.
So I am in communication with the same support team again to get it resolved.
